src/coreclr/jit/valuenum.h

```diff
@@ -359,6 +359,22 @@ class ValueNumStore
 #endif // FEATURE_SIMD
 
 private:
+    // This struct mainly exists to give a place to put the constexpr code for creating
+    // the OpAttribs table. It also gives a scope for the helpers functions used during
+    // initialization.
+    struct VnfOpAttribsType
+    {
+    public:
+        constexpr VnfOpAttribsType();
+        const UINT8& operator[](size_t idx) const { return m_arr[idx]; }
+    private:
+        static constexpr unsigned GetArity(unsigned oper);
+        static constexpr unsigned GetCommutative(unsigned oper);
+        static constexpr unsigned GetFunc(int arity, bool commute, bool knownNonNull, bool sharedStatic);
+
+        UINT8 m_arr[VNF_COUNT];
```

Comment:
```suggestion
        const uint8_t& operator[](size_t idx) const { return m_arr[idx]; }
    private:
        static constexpr unsigned GetArity(unsigned oper);
        static constexpr unsigned GetCommutative(unsigned oper);
        static constexpr unsigned GetFunc(int arity, bool commute, bool knownNonNull, bool sharedStatic);

        uint8_t m_arr[VNF_COUNT];
```
Nit: We want to stick to std C++ type names where possible.